src/stronghold/secret.rs

```diff
@@ -36,31 +38,17 @@ impl SecretManage for StrongholdAdapter {
         _metadata: GenerateAddressMetadata,
     ) -> Result<Vec<Address>> {
         // Stronghold arguments.
```

Comment:
`Error: StrongholdProcedureError(Engine(FatalEngineError("vault `VaultId(TcAMhLekJE6GhvWh6qTYL-T2DKgaUKz7)` does not exist")))`
We get this error now if the key got cleared, and we then want to generate an address, maybe add the following?
```suggestion

        if !self.is_key_available().await {
            return Err(Error::StrongholdKeyCleared);
        }

        // Stronghold arguments.
```

Reply:
Sure